### ISSUE TITLE
Swear replace

### ISSUE THREAD

**Issue by LogGits** [kangarko/ChatControl-Pro#1469]
I want to be able to replace a swear with nothing (via handler) however players say "shit" in chat as the only word and it will make them send an empty chat message. Is there a way to workaround this?

**Comment by lackoSK** [comment]
Hello,

Thank you for your suggestion.

However, I don't think this feature is wanted for everyone. It's not often requested feature and we decided to not add this to ChatControl. 

Thanks!

Regards,
lackoSK

**Comment by kangarko** [comment]
You can actually achieve this by using the Replace_Word key (https://github.com/kangarko/ChatControl-Pro/wiki/Handlers#replace_word) and setting it to

`Replace_Word: " "`

@lackoSK just so you know :)

**Comment by LogGits** [comment]
> You can actually achieve this by using the Replace_Word key (https://github.com/kangarko/ChatControl-Pro/wiki/Handlers#replace_word) and setting it to
> 
> `Replace_Word: " "`
> 
> @lackoSK just so you know :)

@kangarko Thats how i was doing it, the issue is that when a player says a swear as their first message, if there is no message after e.g. the string "shit" if its handled as a swear and i set it to replace swears with `""` then it makes the player send an empty message into chat. 

This is **not** a feature request... It is a question/ potential bug report.

**Comment by lackoSK** [comment]
Hey,

Apology for the late response. Can you explain the issue? 

Thanks

**Comment by LogGits** [comment]
I have a filter that detects swear words. It uses the handler yml file to handle them. I am trying to remove the swear from the message by setting it to `Replace_Word: ""`. This works fine in messages however if a person sends a message that is inly a swear e.g. sending "shit" to to chat, it removes the message and makes the player send a blank message so its just their name with no messege. Like this:

```
Player: shit
```
Turns into:
```
Player: 
```
(Aka senda an empty chat) if you need help fixing this let me know.

**Comment by lackoSK** [comment]
Yeah, and what's the issue? The swear gets replaced by your selected replacer.

**Comment by LogGits** [comment]
Ok ur makin me tilted xD minecraft doesnt let you swnd messages with no text in them. When replace word removes a swear and the whole message was just a swear it makes the player send an empty message (not regular minecraft function). @kangarko u know wat i mean?

**Comment by lackoSK** [comment]
Yeah, I do know what you mean. 

How do you imagine that? How would we do this feature? 
Deny player to chat when the message is empty? - That's it?

**Comment by LogGits** [comment]
Ok so i figured out the issue (but this isnt resolved, i just know what causes it). So in my testing, it appears that the colour code is left there and that allows players to send the message.

So the issue is abit different then i thought. Players can send a message in chat that says `&f` and it will send out this https://gyazo.com/d5044b856bad71a08d6ab909dbc0db8d

Lets take this further. If i send the message `Hello &f &f &f &f &f &f &f &f &f &f &f &f This is a test` this ends up happening: https://gyazo.com/7b88610420707640802d77675c5fbb44

1) Noted in #1453 @kangarko mentioned that chat colours are added before each word. This means when the censor catches the word and replaces it with `""`. It doesn't remove the chatcolour as well.
2) Regarding the sending long spaces issue (`Hello &f &f &f &f &f &f &f &f &f &f &f &f This is a test`) what could be done to prevent that type of spam?

**Comment by lackoSK** [comment]
Hello,

Apologies for the delay. 
- Yes #1453, that's the issue there, which I can't figure out. It always puts &f &f before each word. I'm waiting for kangarko to figure it out - It's bizarre.

Anyways, we can do it like the check like 'if there is more than 3x &f written, we will replace it with 1x &f.'  (Added to my TODO list)

Regards,
lackoSK

**Comment by LogGits** [comment]
I tried resolving this using regex replacer in chatcontrol pro. The issue is that when i test this using regexr.com it works fine but with chatcontrol's match feature it doesnt catch it:
https://gyazo.com/612fdb18873f7d4f9b2ee2739959a99e

Using:
```
match (\&[a-f\d]\s){2,}|(\&[a-f\d]){1,}$
then replace .
```
It didn't even catch the correct stuff... Am i missing something?

**Comment by LogGits** [comment]
@lackoSK its doing that because it hasn't recognised that `{pl_suffix}` provides a chatcolour therefore, defaults to white text. For this, we need a way to not only remove the swear message, but also the chatcolour that the plugin inserts before the word. That will solve the sending empty chat messages.

To solve the other, you should add a method that prevents players from abusing the system to send empty messages via just sending `&f` to chat or `&f &f &l &f` etc.

that will prevent this abuse:
https://gyazo.com/080939b61ab83dc165aefcb9eb404ca1

And then im not sure how @kangarko can will do it but to prevent `&f &f &l &f` abuse in the middle of a sentence to achieve this:
https://gyazo.com/32e2ae0fc9a0cb663dbf15e518572628

**Comment by kangarko** [comment]
Hi,

I apology for a long delay, I was out of office during holidays. The issue with sending empty messages and leftover color codes will be resolved in a few days.

**Comment by kangarko** [comment]
@LogGits apology for the delay.

We were able to fix the issue where the handler would replace the message in its entirety resulting in an empty message. Such messages are no longer sent.

Furthermore the issue with duplicated color codes after each other creating multiple spaces has also been fixed.

We thank you for your patience and welcome you to report more problems should they arise. Have a great day.
